src/Voron/Data/Tables/ZstdLib.cs

```diff
@@ -0,0 +1,279 @@
+                        srcPtr, (UIntPtr)src.Length, 3);
+                }
+                else
+                {
+                    result = ZSTD_compress_usingCDict(_threadCompressContext.Compression, dstPtr,
+                        (UIntPtr)dst.Length, srcPtr, (UIntPtr)src.Length, dictionary.Compression);
+                }
+
+                AssertSuccess(result, dictionary);
+                return (int)result;
+            }
+        }
+
+        public static int Decompress(ReadOnlySpan<byte> src, Span<byte> dst, CompressionDictionary dictionary)
+        {
+            if (_threadCompressContext == null)
+                _threadCompressContext = new CompressContext();
+
+            fixed (byte* srcPtr = src)
+            fixed (byte* dstPtr = dst)
+            {
+                UIntPtr result;
+                if(dictionary == null || dictionary.Compression == null)
+                {
+                    result = ZSTD_decompressDCtx(_threadCompressContext.Decompression, dstPtr, (UIntPtr)dst.Length, srcPtr, (UIntPtr)src.Length);
+                }
+                else
+                {
+                    result = ZSTD_decompress_usingDDict(_threadCompressContext.Decompression, dstPtr, (UIntPtr)dst.Length, srcPtr, (UIntPtr)src.Length, dictionary.Decompression);
+                }
+                AssertSuccess(result, dictionary);
+                return (int)result;
+            }
+        }
+
+        public class CompressionDictionary : IDisposable
+        {
+            public void* Compression;
+            public void* Decompression;
+            public int Id;
+            public byte ExpectedCompressionRatio;
+
+#if DEBUG
+            public string DictionaryHash;
+#endif
+
+            public CompressionDictionary(int id, byte* buffer, int size, int compressionLevel)
+            {
+                Id = id;
+                if (buffer == null)
+                {
+#if DEBUG
+                    DictionaryHash = "<null>";
+#endif
+                    return;
+                }
+
+#if DEBUG
+                var hash = stackalloc byte[32];
+                Sodium.crypto_generichash(hash, (UIntPtr)32, buffer, (ulong)size, null, UIntPtr.Zero);
+                DictionaryHash = Convert.ToBase64String(new ReadOnlySpan<byte>(hash, 32));
+#endif
+
+                Compression = ZSTD_createCDict(buffer, (UIntPtr)size, compressionLevel);
+                Decompression = ZSTD_createDDict(buffer, (UIntPtr)size);
+                if (Compression == null || Decompression == null)
+                {
+                    Dispose();
+                    throw new OutOfMemoryException("Unable to allocate memory for dictionary");
+                }
+            }
+
+            public override string ToString()
+            {
+                return "Dic #" + Id 
+#if DEBUG
+					 + " - " + DictionaryHash
+#endif
+
+;
+            }
+
+            ~CompressionDictionary()
+            {
```

Comment:
Maybe let's add some logging or just console write line in DEBUG when the finalizer is called